File: demo2/src/main/java/org/mosquito/demo2/service/IOrderService.java
package org.mosquito.demo2.service;

import org.mosquito.demo2.dto.OrderExtDto;
import org.mosquito.framework.jdbc.service.IBaseService;
import org.mosquito.demo2.model.Order;
import org.mosquito.demo2.dto.OrderDto;

import java.util.List;

/**
 * 业务类接口
 *
 * @author jcchen
 * @Date 2018-5-27
 * 
 */
public interface IOrderService extends IBaseService<OrderDto, Order>{

    List<OrderExtDto> findListExt(OrderDto dto);

}

File: demo3/src/main/scala/org/mosquito/demo3/service/IOrderService.scala
package org.mosquito.demo3.service

import java.util

import org.mosquito.demo3.dto.{OrderDto, OrderExtDto}
import org.mosquito.demo3.model.Order
import org.mosquito.framework.jdbc.service.IBaseService

trait IOrderService extends IBaseService[OrderDto, Order] {
  def findListExt(dto: OrderDto): util.List[OrderExtDto]
  def findListExt2(dto: OrderDto): Seq[OrderExtDto]
  def selectAll2(dto: OrderDto): util.List[OrderDto]
}
